### ISSUE TITLE
FIDO2 security key prompt for passkey _registration_ does not appear

### ISSUE THREAD

**Issue by FID02** [GrapheneOS/os-issue-tracker#3347]
On GrapheneOS, registering a FIDO2 passkey on a security key is currently not possible on most websites, due to a missing registration prompt.

_Background:_ 
Google Play Services added [support for PIN with FIDO2 in September. ](https://support.google.com/product-documentation/answer/11412553?hl=en#zippy=%2Cseptember): "[Phone] Adding Pin Protocol support for Fido2 on Android Platform."

_Issue description:_

On stock PixelOS using Chrome, when attempting to register a passkey on a website, the user is presented with a dialog offering to save a passkey to Google Password Manager. However, the user is also offered the option to "Use a different device" which, when pressed, further offers the option to save the passkey on a USB security key. This works fine on PixelOS. 

Screenshot from Chrome on PixelOS:

![image](https://github.com/GrapheneOS/os-issue-tracker/assets/99329251/6488b352-ebb8-46bf-a0a8-c70f9562e16d)

On GrapheneOS using Vanadium, the dialog that offers to save a passkey to Google Password Manager is also shown. However, the option to "Use a different device" is _not_ present. The dialog otherwise appears identical to the dialog shown on PixelOS.

Screenshot from Vanadium:

![image](https://github.com/GrapheneOS/os-issue-tracker/assets/99329251/a74b1899-4a6d-4635-a1a1-980551f25107)

A notable exception is microsoft.com which, when the user selects to store a passkey on a security key, the Google Password Manager dialog is not shown but skips directly to the USB security key prompt. In Vanadium, saving a passkey for microsoft.com on a security key _works fine_, and the passkey on the security key can then be used to sign in to microsoft.com on GrapheneOS _without issues_.

I saved logs from both Play Services and GSF immediately after triggering the dialog in Vanadium (from the second screenshot). Logs are attached here, although unclear to me if these are in any way helpful:
[Google Play services log 7c2d0f845e7c.txt](https://github.com/GrapheneOS/os-issue-tracker/files/14623735/Google.Play.services.log.7c2d0f845e7c.txt)
[Google Services Framework log b921c8d5a686.txt](https://github.com/GrapheneOS/os-issue-tracker/files/14623736/Google.Services.Framework.log.b921c8d5a686.txt)

Stock PixelOS version AP1A.240305.019.A1 tested on a P6a with Chrome 122.0.6261.106.
GrapheneOS version 2024031400 tested on a P8 with Vanadium 123.0.6312.40 and Play Services version 24.08.12.

**Comment by FID02** [comment]
I have found a workaround for this in Vanadium. It requires changing a specific chrome://flags setting called 'Android Credential Management for passkeys' from _Default_ to either _Enabled for 3rd party passkeys_ or _Enabled for Google Password Manager and 3rd party passkeys_.

Change that flag, relaunch Vanadium, and you can now store passkeys on your security key.

Of course, changing that flag causes other breakages in Vanadium, so this is only to demonstrate that it _does_ work and is possible to work, and there doesn't appear to be an issue with Sandboxed Google Play.

It appears that you need Play Services 24.22.13 or newer in order to test this. I have tested this on GrapheneOS [2024061200](https://grapheneos.org/releases#2024061200) on both a Pixel 8 and Pixel 6a.

Notably, this works out of the box in both Chrome and Brave, on GrapheneOS, without having to change any settings at all. Which _might_ point towards an issue in Vanadium (but this is me making an uneducated guess based on my observations so far).

---

You can test storing passkeys on a security key by going to https://passkey.org and signing up for an account there. You don't need to have a security key in order to trigger the menus.

Screenshots of the passkey menus that are shown when the flag is set in chrome://flags in Vanadium:

When you are not signed in to the Play Store:
![image](https://github.com/GrapheneOS/os-issue-tracker/assets/99329251/bd87c009-b803-44ea-b216-69bf02a66cd1)

When you are signed in to the Play Store:
![image](https://github.com/GrapheneOS/os-issue-tracker/assets/99329251/37c7438c-1e1f-4ee6-b2e7-cc53c5fd50d5)

**Comment by FID02** [comment]
Ah, if you mean Vanadium version 126.0.6478.50.1 and Vanadium Config 23 then I will try with them as soon as the next QPR3-based release is pushed to Alpha.

**Comment by FID02** [comment]
> @FID02 Please try with the latest Vanadium and Vanadium Config with a fresh profile. I thought we were already setting the default to that value.

The issue persists with Vanadium version 126.0.6478.50.1 and Vanadium Config 23. I have tested in a completely fresh profile, with Play Services 24.22.32 and Vanadium settings set to defaults, no settings changed. Tried in two fresh profiles, put back to rest and then unlocked again a few times. Pixel 8.

Also persists on a Pixel 6a that was factory reset. Same versions and configuration as on the Pixel 8.

**Comment by freijon** [comment]
It seems the chrome://flags setting called 'Android Credential Management for passkeys' no longer exists. Are there other workarounds?

**Comment by freijon** [comment]
> Use the standard autofill configuration

You mean in the Vanadium settings? I tried it - no luck. I'm trying to use the Open Passkey Authenticator (OPA) instead of Google

**Comment by freijon** [comment]
Yes, and I get stuck in a registering loop where I have to enter the fingerprint again and again
